src/webots/gui/WbMainWindow.cpp

```diff
@@ -1639,11 +1639,17 @@ void WbMainWindow::upload(char type) {
 
   QNetworkReply *reply = WbNetwork::instance()->networkAccessManager()->post(request, multiPart);
 
-  mUploadProgressDialog = new QProgressDialog(tr("Uploading on %1...").arg(uploadUrl), "Cancel", 0, 100, this);
+  QString cancelText = "cancel";
+  mUploadProgressDialog = new QProgressDialog(tr("Uploading on %1...").arg(uploadUrl), cancelText, 0, 100, this);
   mUploadProgressDialog->setWindowTitle(tr("%1").arg(uploadUrl));
   mUploadProgressDialog->show();
   connect(reply, &QNetworkReply::uploadProgress, this, &WbMainWindow::updateUploadProgressBar);
 
+  QPushButton *pushButtonCancel = new QPushButton(mUploadProgressDialog);
+  mUploadProgressDialog->setCancelButton(pushButtonCancel);
+  mUploadProgressDialog->setCancelButtonText(cancelText);
+  connect(pushButtonCancel, &QPushButton::pressed, reply, [reply] { reply->abort(); });
```

Comment:
Is the `mUploadProgressDialog->setCancelButtonText(cancelText);` really needed?
```suggestion
  QPushButton *cancelButton = new QPushButton(mUploadProgressDialog);
  mUploadProgressDialog->setCancelButton(cancelButton);
  mUploadProgressDialog->setCancelButtonText(cancelText);
  connect(cancelButton, &QPushButton::pressed, reply, [reply] { reply->abort(); });
```

Reply:
I have changed the name, but it seems necessary to keep `mUploadProgressDialog->setCancelButtonText(cancelText);` as the default `Cancel` text is removed when modifying the buttons functionality with `mUploadProgressDialog->setCancelButton(cancelButton);`